### PR TITLE
Rework and extend Round Results Page

### LINKED ISSUE #1455: Create Round Results page
**User Story:**
As a program manager
I want to view and confirm my round results
So that I can finalize and payout my round

**Acceptance Criteria**
GIVEN that I am on my round's details page AND my round has ended
WHEN I click on Round Results tab
THEN I see the Finalize Round tab
AND the Vote Coefficients and Matching Distribution sections

GIVEN that I am on the Finalize Round tab AND my round has ended
WHEN I look at the Matching Distribution section
THEN I see the current calculated distribution for the round

GIVEN that I am on the Finalize Round tab AND my round has ended
WHEN I click on the Vote Coefficient or Matching Distribution CSV download button
THEN that csv file is downloaded to my machine

**Product & Design Links:**
[Figma](https://www.figma.com/file/V113F2Pp5m2qDM5odXjfFu/Manager-S17?node-id=892-53042&t=UcZwCdJ36qheVEp4-4)
NOTE: this doesn't include adding the raw round data tab or the saturation / revision flow

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by vacekj**
fixes #1455 

todo:
 
- [x] Format Finalize Round tab to match design
- [ ] Round and Project matching stats in the allo indexer
- [x] tests

### CODE REVIEW

**Comment by boudra** [comment]
I noticed you're installing `papaparse`. We already have CSV libraries installed, check the `package.json`.

**Comment by vacekj** [comment]
> I noticed you're installing `papaparse`. We already have CSV libraries installed, check the `package.json`. Also I don't think we need to parse CSV in the front end to implement this feature but maybe I am missing context.

Removed papaparse in favour of csv-parse which we already had installed (can be switched back to papaparse, but only one should be used at a time). 

CSV Parsing isn't necessary for the ACs in #1455, @hmrtn was your intention to bundle all the stories for this page into this PR? I would advise against this, for reasons previously discussed, and instead keep the 1:1:1 Ticket - Branch - PR rule. I see that some of the ACs from the other tickets are already ticked off in this PR, so please update the tickets to mention the ACs were moved here.

**Comment by hmrtn** [comment]
In order to proceed with download and tables we need the matching calculation handlers in the allo indexer client.

**Comment by hmrtn** [comment]
NOTE: This hardcodes `isBeforeRoundEndTime = false`. To resolve this, changes are required in the indexer. Namely the client `Round type` does not implement the `roundStart/Endtime`.
This is necessary to merge as the tests will fail otherwise.

**Comment by hmrtn** [comment]
Also see the TODO's. More tickets will be needed to close these out - related to uploading, finalizing, etc //

File: packages/round-manager/.run/RM All Tests.run.xml
Comment (boudra) [inline packages/round-manager/.run/RM All Tests.run.xml:1]: what is this file? is it an IDE file?

File: packages/round-manager/src/features/round/ViewRoundResults_old.tsx
Comment (boudra) [inline packages/round-manager/src/features/round/ViewRoundResults_old.tsx:1]: we shouldn't have these _old files

**Review by boudra — CHANGES_REQUESTED** [review]

File: packages/round-manager/.run/RM All Tests.run.xml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="RM All Tests" type="JavaScriptTestRunnerJest">
     <node-interpreter value="project" />
-    <node-options value="" />
-    <jest-package value="$PROJECT_DIR$/packages/round-manager/node_modules/react-scripts" />
+    <jest-package value="$PROJECT_DIR$/packages/round-manager/node_modules/@craco/craco" />
     <working-dir value="$PROJECT_DIR$/packages/round-manager" />
     <envs />
     <scope-kind value="ALL" />
```
Comment (gravityblast) [inline packages/round-manager/.run/RM All Tests.run.xml:7]: we need to remove this JetBrain file

File: packages/round-manager/package.json [hunk 2]
```diff
@@ -46,7 +46,7 @@
     "@testing-library/user-event": "^14.1.1",
     "@types/jest": "^27.4.1",
     "@types/node": "^17.0.25",
-    "allo-indexer-client": "github:gitcoinco/allo-indexer-client#62b3dcefa5e6a5c9f191c104780b26930067ec40",
+    "allo-indexer-client": "github:gitcoinco/allo-indexer-client#61d050e82e58cc8e55f79faf8fbb3d0236d8ff50",
```
Comment (gravityblast) [inline packages/round-manager/package.json:49]: let's create a PR on that repo as well

**Comment by boudra** [comment]
before we merge let's make sure the round stats work using some of the current mainnet rounds

File: packages/round-manager/package.json (hunk 2 above)
Comment (gravityblast) [inline packages/round-manager/package.json:49]: ah I see this is already in main, let's create a new PR if we need more changes

**Comment by boudra** [comment]
> NOTE: This hardcodes `isBeforeRoundEndTime = false`. To resolve this, changes are required in the indexer. Namely the client `Round type` does not implement the `roundStart/Endtime`.
> This is necessary to merge as the tests will fail otherwise. 

is something blocking you from doing this straight away? the allo indexer client can be updated and pushed in a branch, then we update the commit hash here

**Comment by hmrtn** [comment]
here is the allo indexer commit in the latest changes: https://github.com/gitcoinco/allo-indexer-client/commit/52be896010b22083390ca3ec6e05da9f97a4dd19

File: packages/round-manager/.run/RM All Tests.run.xml
Comment (hmrtn) [reply packages/round-manager/.run/RM All Tests.run.xml:1]: yes - jetbrains things

Comment (boudra) [reply packages/round-manager/.run/RM All Tests.run.xml:1]: removed

File: packages/round-manager/src/features/round/ViewRoundResults_old.tsx
Comment (boudra) [reply packages/round-manager/src/features/round/ViewRoundResults_old.tsx:1]: removed

File: packages/round-manager/.run/RM All Tests.run.xml (hunk 1 above)
Comment (boudra) [reply packages/round-manager/.run/RM All Tests.run.xml]: removed

**Review by boudra — APPROVED** [review]

**Comment by vacekj** [comment]
image
you love to see it 🔥

**Pull request merged**